### PR TITLE
Remove calicoctl from libcalico-go

### PR DESCRIPTION
**Pull request by robbrockbank**
Remove calicoctl
-  Remove make targets for building calicoctl
-  Remove calicoctl specific test files

In terms of reviewing this:
-  Validate that you can't see anything wrong conceptually with removing calicoctl and having as a temporary branch in calico-containers.
-  Validate that the README makes that clear
-  Double check the Makefile.

Otherwise most of it is removal of files.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -86,9 +49,28 @@ run-etcd:
...
 clean:
 	find . -name '*.coverprofile' -type f -delete
-	rm -rf bin \
-	       release \
-	       vendor
+	rm -rf vendor
+
+.PHONY: help
+## Display this help text
+help: # Some kind of magic from https://gist.github.com/rcmachado/af3db315e31383502660
```
Comment (gunjan5) [inline Makefile]: not major but I get:
```bash
wc: illegal option -- L
usage: wc [-clmw] [file ...]
```
when I do `make run` on my mac. Full output:
```bash
$ make help
Available targets
wc: illegal option -- L
usage: wc [-clmw] [file ...]
ut: Run the UTs locally.  This requires a local etcd to be running.
test-containerized: Run the tests in a container. Useful for CI, Mac dev.
run-etcd: Run etcd as a container
help: Display this help text
```

File: README.md [hunk 2]
```diff
@@ -4,48 +4,30 @@
 This repositiory contains Calico's Go components:
 
 - `libcalico`, which can be imported as `"github.com/projectcalico/libcalico-go/lib"`
-- `calicoctl`
 
-## Common set-up
+This library is used by both `calicoctl` and `felix`.
```
Comment (gunjan5) [inline README.md]: should we also mention `calico-cni` here?

File: README.md [hunk 3]
```diff
@@ -4,48 +4,30 @@
...
-cd $GOPATH/src/github.com/projectcalico/libcalico-go
-```
+If you are developing against the code in libcalico, please run the tests before
+submitting a Pull Request.
 
-- Populate the `vendor/` directory in the project's root with this project's dependencies:
-```
-glide install
-```
+To run the tests locally (requires a full golang environment):
```
Comment (gunjan5) [inline README.md]: Might be worth mentioning Go 1.7+ minimum

File: README.md [hunk 4]
```diff
@@ -4,48 +4,30 @@
...
-### Release build
-
-For releases, we use a Docker-based build to ensure a clean environment with an appropriate glibc.  Specifically, we use a CentOS 6.6 container image to build against glibc v2.12.  this ensures compatibility with any later glibc.
-
-To do a release build, run:
-```
-make release/calicoctl
-```
-The binary will be emitted to `./releases/calicoctl-<version>`
+    make test-containerized
```
Comment (gunjan5) [inline README.md]: This target still doesn't work, but we can chat about it later.

File: Makefile (hunk 1 above)
Comment (robbrockbank) [inline Makefile]: I've just fixed the width rather than try to calculate it.

File: README.md (hunk 2 above)
Comment (robbrockbank) [inline README.md]: sure

File: README.md (hunk 3 above)
Comment (robbrockbank) [inline README.md]: sure

File: README.md (hunk 4 above)
Comment (robbrockbank) [inline README.md]: I've merged in that fix.   Tests were working for me though even without that - but it's good to not leave the vendor directory permissions in a mess.

**Pull request merged**